python/ql/lib/semmle/python/frameworks/Stdlib.model.yml

```
@@ -21,9 +21,161 @@ extensions:
       - ['argparse.ArgumentParser', 'Member[parse_args,parse_known_args].WithArity[0].ReturnValue', 'commandargs']
 
       - ['os', 'Member[read].ReturnValue', 'file']
+
+  - addsTo:
+      pack: codeql/python-all
+      extensible: sinkModel
+    data:
+      - ["zipfile.ZipFile","Member[extractall].Argument[0,path:]", "path-injection"]
+
   - addsTo:
       pack: codeql/python-all
       extensible: summaryModel
     data:
-      - ['argparse.ArgumentParser', 'Member[parse_args,parse_known_args]', 'Argument[0,args:]', 'ReturnValue', 'taint']
+      # See https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser
+      - ["argparse.ArgumentParser", "Member[_parse_known_args,_read_args_from_files]", "Argument[0,arg_strings:]", "ReturnValue", "taint"]
       # note: taint of attribute lookups is handled in QL
+      - ["argparse.ArgumentParser", "Member[parse_args,parse_known_args]", "Argument[0,args:]", "ReturnValue", "taint"]
```

I don't think my original comment was super clear, so here's a suggestion to clean that up a bit :sweat_smile: (found QL code [here](https://github.com/github/codeql/blob/7483075b7eeb577c5bedb43a9ce3b6a37e427dc2/python/ql/lib/semmle/python/frameworks/Stdlib.qll#L5005-L5020) for reference)

```suggestion
      # See https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser
      # note: taint flow for attribute lookups on `argparse.ArgumentParser` is handled in QL
      - ["argparse.ArgumentParser", "Member[_parse_known_args,_read_args_from_files]", "Argument[0,arg_strings:]", "ReturnValue", "taint"]
      - ["argparse.ArgumentParser", "Member[parse_args,parse_known_args]", "Argument[0,args:]", "ReturnValue", "taint"]
```

---

Ah, thanks, I was a bit puzzled by that comment; it makes much more sense now :-)